src/features/common/Layout/UserPropsContext.tsx

```diff
@@ -112,29 +112,49 @@ function UserPropsProvider({ children }: any): ReactElement {
     }
   }, [isImpersonationModeOn]);
 
+  const value: UserPropsContextInterface | null = useMemo(
+    () => ({
+      contextLoaded,
+      setContextLoaded,
+      token,
+      setToken,
+      user: profile,
+      setUser,
+      userLang,
+      setUserLang,
+      isImpersonationModeOn,
+      setIsImpersonationModeOn,
+      isLoading,
+      isAuthenticated,
+      loginWithRedirect,
+      logoutUser,
+      auth0User: user,
+      auth0Error: error,
+      loadUser,
+    }),
+    [
+      contextLoaded,
+      token,
+      profile,
+      userLang,
+      isImpersonationModeOn,
+      isLoading,
+      isAuthenticated,
+      user,
+      error,
+    ]
+  );
```

Comment:
This isn't an expensive operation, so perhaps it isn't necessary to have a `useMemo`. You could assign the value to the context provider directly. 

An added advantage is that you will see a warning if there is a difference in the interface and the value you're composing for the provider.